Added snippets, images, notes, clarifying statements, clean up, etc.

IEvangelist opened the pull request:
## Summary

- Update _.github/CODEOWNERS_ and _docfx.json_ files.
- Add additional steps
- Add new images
- Add snippets and code sample projects / solution
- Add clarifying statements
- Add notes and additional context where needed
- Add xref links to API
- Fix bug in existing sample code
- Slightly modify generated source
- Update analyzer packages to latest stable

Fixes #27119

✔️ [Preview: Source Generators](https://review.docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview?branch=pr-en-us-27594)

IEvangelist committed:
Added snippets, images, notes, clarifying statments, clean up, etc.

adegeo reviewed (approved):
Looks good. Though the folder names in `./media` and `./snippets` is supposed to be the same name as the matching .md file: `/source-generators-overview/`

adegeo reviewed (changes requested):
Just spotted a few tiny things, updating this review

adegeo commented on docs/csharp/roslyn-sdk/source-generators-overview.md:
````diff
@@ -1,93 +1,75 @@
-        static partial void HelloFrom(string name);
-    }
-    ```
+    :::code language="xml" source="snippets/source-generators/ConsoleApp/Program.cs":::
````
Code language is wrong
```suggestion
    :::code language="csharp" source="snippets/source-generators/ConsoleApp/Program.cs":::
```

IEvangelist replied:
Already fixed

adegeo commented on docs/csharp/roslyn-sdk/source-generators-overview.md:
````diff
@@ -1,93 +1,75 @@
 
-    ```xml
-    <Project Sdk="Microsoft.NET.Sdk">
+    :::code language="xml" source="snippets/source-generators/SourceGenerator/SourceGenerator.csproj":::
````
Code language is wrong
```suggestion
    :::code language="csharp" source="snippets/source-generators/SourceGenerator/SourceGenerator.csproj":::
```

IEvangelist replied:
The _.csproj_ file is XML, this suggestion is incorrect 😄

adegeo replied:
lol

adegeo commented on docs/csharp/roslyn-sdk/source-generators-overview.md:
````diff
@@ -102,51 +84,48 @@ In this guide, you'll explore the creation of a source generator using the <xref
-    ```
+1. Replace the contents of the <xref:Microsoft.CodeAnalysis.ISourceGenerator.Execute%2A?displayProperty=nameWithType> method, with the following implementation:
+
+    :::code source="snippets/source-generators/SourceGenerator/HelloSourceGenerator.cs":::
````
Missing language
```suggestion
    :::code language="csharp" source="snippets/source-generators/SourceGenerator/HelloSourceGenerator.cs":::
```

IEvangelist replied:
The language is inferred for well-known types, for example; files with the _.cs_ extension are automatically mapped as C#.

adegeo commented on docs/csharp/roslyn-sdk/source-generators-overview.md:
```diff
@@ -102,51 +84,48 @@ In this guide, you'll explore the creation of a source generator using the <xref
+    From the `context` object we can access the compilations' entry point, or `Main` method. The `mainMethod` instance is an <xref:Microsoft.CodeAnalysis.IMethodSymbol>, and it represents a method or method-like symbol (including constructor, destructor, operator, or property/event accessor). From this object we can reason about the containing namespace (if one is present) and the type. The `sour ...
+
+    > [!TIP]
+    > The `hintName` parameter from the <xref:Microsoft.CodeAnalysis.GeneratorExecutionContext.AddSource%2A?displayProperty=nameWithType> method can be any unique name, however; it is common to provide an explicit C# file extension such as `".g.cs"` or `".generated.cs"`. This will cause the generated file name to further help identify itself as being source generated.
```
nit: I think readability (acro suggestions) this can be improved a little
```suggestion
    > The `hintName` parameter from the <xref:Microsoft.CodeAnalysis.GeneratorExecutionContext.AddSource%2A?displayProperty=nameWithType> method can be any unique name. It's common to provide an explicit C# file extension such as `".g.cs"` or `".generated.cs"` for the name. The file name helps identify the file as being source generated.
```

IEvangelist committed:
A few tweaks and modifications

adegeo commented on docs/csharp/roslyn-sdk/source-generators-overview.md:
````diff
@@ -1,93 +1,87 @@
-        static partial void HelloFrom(string name);
-    }
-    ```
+    :::code source="snippets/source-generators/ConsoleApp/Program.cs":::
````
Missing language
```suggestion
    :::code language="csharp" source="snippets/source-generators/ConsoleApp/Program.cs":::
```

adegeo reviewed (changes requested):
Looks good. Though the folder names in `./media` and `./snippets` is supposed to be the same name as the matching .md file: `/source-generators-overview/`

Some other quick fixes you can do with the code references.

IEvangelist committed:
Update docs/csharp/roslyn-sdk/source-generators-overview.md

Co-authored-by: Andy (Steve) De George <67293991+adegeo@users.noreply.github.com>

IEvangelist commented:
> Looks good. Though the folder names in `./media` and `./snippets` is supposed to be the same name as the matching .md file: `/source-generators-overview/`
> 
> Some other quick fixes you can do with the code references.

Thanks @adegeo for the quick review. I've addressed all of your feedback, except for the folder naming - there was already a media directory with _source-generators_ so I kept that, and when I created the new snippets directory, I opted to name it the same.

adegeo commented:
> Thanks @adegeo for the quick review. I've addressed all of your feedback, except for the folder naming - there was already a media directory with _source-generators_ so I kept that, and when I created the new snippets directory, I opted to name it the same.

@IEvangelist Just FYI we can use `_#` appended to the folder to keep it associated with the article when we have more than one full set of snippets.

```
./snippets/source-generators-overview/
./snippets/source-generators-overview_1/
./snippets/source-generators-overview_2/
```
